### PR TITLE
ocaml: update to 4.14.2

### PR DESCRIPTION
**Pull request by jiegec**
Topic Description
-----------------

- ocaml-augeas: bump REL due to ocaml update to 4.14.2
- lablgtk: drop
- gtkmathview: drop
- gmetadom: drop
- zarith: bump REL due to ocaml update to 4.14.2
- pdfsandwich: bump REL due to ocaml update to 4.14.2
- opam: bump REL due to ocaml update to 4.14.2
- ocamlbuild: bump REL due to ocaml update to 4.14.2
- ocaml-stdlib-shims: bump REL due to ocaml update to 4.14.2
- ocaml-result: bump REL due to ocaml update to 4.14.2

... and 25 more commits

Package(s) Affected
-------------------

- camlp-streams: 5.0.1
- camlp4: 4.14+1
- camlp5: 8.02.01
- dune: 3.4.1-2
- facile: 1.1.3-4
- findlib: 1.9.5-3
- hevea: 2.36-1
- hivex: 1.3.23-2
- kalzium: 23.08.5-1
- libguestfs: 1.52.0-2
- ocaml-astring: 0.8.5
- ocaml-augeas: 0.6-1
- ocaml-bos: 0.2.1
- ocaml-cmdliner: 1.2.0
- ocaml-csexp: 1.5.1-1
- ocaml-fmt: 0.9.0
- ocaml-fpath: 0.7.3
- ocaml-graphics: 5.1.0-2
- ocaml-integers: 0.4.0-1
- ocaml-logs: 0.7.0
- ocaml-num: 1.3-4
- ocaml-pp: 1.1.2-1
- ocaml-re: 1.11.0
- ocaml-result: 1.5-2
- ocaml-stdlib-shims: 0.2.0-2
- ocaml-topkg: 1.0.7
- ocaml: 4.14.2
- ocamlbuild: 0.14.1-2
- opam: 2.1.3-2
- pdfsandwich: 0.1.7-3
- zarith: 1.9.1-4

Security Update?
----------------

No

Build Order
-----------

```
#buildit ocaml:-pkgbreak findlib ocaml-pp ocaml-result ocaml-csexp dune camlp-streams ocamlbuild camlp4 opam ocaml-topkg ocaml-astring ocaml-fpath ocaml-cmdliner ocaml-fmt ocaml-logs ocaml-bos ocaml-re camlp5 facile hevea hivex kalzium ocaml-augeas libguestfs ocaml-graphics ocaml-integers ocaml-num ocaml-stdlib-shims pdfsandwich zarith ocaml
```

Test Build(s) Done
------------------

**Primary Architectures**

- [x] AMD64 `amd64`
- [x] AArch64 `arm64`
- [x] LoongArch 64-bit `loongarch64`

**Secondary Architectures**

- [x] Loongson 3 `loongson3`
- [x] PowerPC 64-bit (Little Endian) `ppc64el`
- [x] RISC-V 64-bit `riscv64`

**Experimental Architectures**

- [ ] MIPS R6 64-bit (Little Endian) `mips64r6el`

### CODE REVIEW

File: lang-ocaml/ocaml-rresult/autobuild/build [hunk 1]
```diff
@@ -0,0 +1,5 @@
+abinfo "Building ocaml-rresult ..."
+ocaml pkg/pkg.ml build --dev-pkg false
```
Comment (MingcongBai) [inline lang-ocaml/ocaml-rresult/autobuild/build]: Absolute path for `pkg/pkg.ml`?

File: lang-ocaml/ocaml-logs/autobuild/build [hunk 2]
```diff
@@ -0,0 +1,5 @@
+abinfo "Building ocaml-logs ..."
+ocaml pkg/pkg.ml build --dev-pkg false --with-js_of_ocaml false --with-lwt false
```
Comment (MingcongBai) [inline lang-ocaml/ocaml-logs/autobuild/build]: Absolute path for `pkg/pkg.ml`?

File: lang-ocaml/ocaml-fpath/autobuild/build [hunk 3]
```diff
@@ -0,0 +1,5 @@
+abinfo "Building ocaml-fpath ..."
+ocaml pkg/pkg.ml build --dev-pkg false
```
Comment (MingcongBai) [inline lang-ocaml/ocaml-fpath/autobuild/build]: Absolute path for `pkg/pkg.ml`?

Comment (MingcongBai) [reply lang-ocaml/ocaml-fpath/autobuild/build]: Also, possible candidate for a template?

File: lang-ocaml/ocaml-topkg/autobuild/build [hunk 4]
```diff
@@ -0,0 +1,7 @@
+abinfo "Building ocaml-topkg ..."
+ocaml pkg/pkg.ml build --pkg-name topkg
```
Comment (MingcongBai) [inline lang-ocaml/ocaml-topkg/autobuild/build]: Absolute path for `pkg/pkg.ml`?

**Review by MingcongBai — CHANGES_REQUESTED** [review]

File: lang-ocaml/ocaml-rresult/autobuild/build (hunk 1 above)
Comment (jiegec) [reply lang-ocaml/ocaml-rresult/autobuild/build]: Fixed

File: lang-ocaml/ocaml-logs/autobuild/build (hunk 2 above)
Comment (jiegec) [reply lang-ocaml/ocaml-logs/autobuild/build]: Fixed

File: lang-ocaml/ocaml-fpath/autobuild/build (hunk 3 above)
Comment (jiegec) [reply lang-ocaml/ocaml-fpath/autobuild/build]: Fixed

File: lang-ocaml/ocaml-topkg/autobuild/build (hunk 4 above)
Comment (jiegec) [reply lang-ocaml/ocaml-topkg/autobuild/build]: Fixed

**Review by MingcongBai — APPROVED** [review]

**Pull request merged**